mergesort.rb

```diff
@@ -1,24 +1,67 @@
-def mergesort(a)
-    # if the array size is 0|1 then the list is considered sorted, return sorted array
-    # split the list into two halves
-    # merge sort each half
-    # combine the sorted halves
+# precondition: a and b are sorted
+def combine(a, b)
+   result = []
+
+   while a.size > 0 && b.size > 0
+     result << if a[0] <= b[0]
+       a.shift
+     else
+       b.shift
+     end
+   end
+
+   result.concat(a).concat(b)
 end
 
+
+
+
+
+# def combine(a, b)
+#   print a
+#   # create a results array
+#   results_array = []
+#   # counters pointing to the index of the smallest elements in each array
+#   while a.size > 0 && b.size > 0
+#     if a.shift < b.shift
+#
+#       results_array << a
+#     else
+#       results_array << b
+#     end
+#   results_array
+#   end
+#   # check that we have elements to compare
+#   # push the smaller element onto the result array
+#   # if there are elements left over in a, move them to result
+#   # if there are elements left over in b, move them to result
+# end
+
+
 def split_array(a)
-    # find the middle index
-    # split the array in half - take a look at take and drop in the ruby docs, or use ranges
-    # return left and right halves of array as separate arrays - did you know a method can return multiple arrays? look it up
+  # find the middle index
+  # a.each_slice( (a.size/2.0).round ).to_a
+  # split the array in half - take a look at take and drop in the ruby docs, or use ranges
+  # return left and right halves of array as separate arrays - did you know a method can return multiple arrays? look it up
+  left, right = a.each_slice((a.size/2.0).round).to_a
+
+  return  left, right
+
```

Comment:
It looks like your split_array function works, but it's not immediately obvious that line 46 is correct. In general, you don't get extra points for writing fewer lines of code. Code will written once but read many, many times, so even though it may take you a little more time to write a more readable solution, it will greatly help your fellow programmers. 

(There are "code golf" competitions where you do, in fact, get extra points for writing fewer lines of code, if that's your thing!)